src/main.cpp

```diff
@@ -818,15 +821,18 @@ int main(int argc, char* argv[])
     if (!is_root())
       return 1;
 
-    if (args.search.find(':') == std::string::npos &&
-        (args.search.find("struct") == 0 || args.search.find("union") == 0 ||
-         args.search.find("enum") == 0)) {
+    if (is_type_name(args.search)) {
       // Print structure definitions
       bpftrace.parse_btf({});
       bpftrace.probe_matcher_->list_structs(args.search);
       return 0;
     }
 
+    if (args.search.find(".") != std::string::npos) {
+      LOG(WARNING) << "It appears that \'" << args.search
```

Comment:
Do we have probes that contain `.` ? I'm just wondering why not exit here with an `ERROR` or `ABORT`.

Reply:
We do, e.g. `uprobe:./a.out:*`. Now thinking about it, we should go with a stricter condition: contains `.` and does not contain neither `:` of `*`. Then, we can probably go with `LOG(ABORT)`, too.

Reply:
In the end, I decided to stay on the safe side and continue with the execution, so kept `LOG(WARNING)` there.

Reply:
Then maybe updating the warning message slightly would be good e.g. 
"It appears that X is a filename but the file does not exist. Proceeding as if X is a search pattern".

Reply:
Good idea, I adjusted it a bit to be more compact:
```
WARNING: It appears that 'script.bt' is a filename but the file does not exist. Treating 'script.bt' as a search pattern.
```